osm2streets/src/output.rs

```diff
@@ -89,52 +76,90 @@ impl StreetNetwork {
 
     // TODO get_designations -> Vec<Designation> {...} // travel areas, parking, etc.
 
-    // TODO pub fn get_markings -> Vec<Marking> {...}
-
-    /// Generate painted areas.
-    pub fn get_paint_areas(&self) -> Vec<PaintArea> {
-        // TODO generate semantic markings first in get_markings, then render them into areas here.
-
-        let mut output = Vec::new();
+    /// Generate markings, described semantically.
+    pub fn get_markings(&self) -> Vec<Marking> {
+        let mut markings = Vec::new();
 
         for road in self.roads.values() {
             // Always oriented in the direction of the road
             let mut lane_centers = road.get_lane_center_lines();
-
+            let guess_overtaking = match road.highway_type.as_str() {
+                "motorway" | "trunk" | "primary" => false,
+                _ => true,
+            };
+
+            // Add the left road edge.
+            if let Some(first_lane) = road.lane_specs_ltr.first() {
+                if matches!(
+                    first_lane.lt.to_traffic_mode(),
+                    Some(TrafficMode::Bike) | Some(TrafficMode::Motor)
+                ) {
+                    if let Ok(edge_line) = lane_centers[0].shift_left(first_lane.width / 2.0) {
+                        markings.push(Marking::longitudinal(
+                            edge_line,
+                            LaneEdgeKind::edge(),
+                            [LaneType::Buffer(BufferType::Verge), first_lane.lt],
+                        ));
+                    }
+                }
+            }
+            // Add longitudinal markings between lanes.
```

Comment:
It is so much nicer to describe the semantics in a layer first, and get to rendering details later!